src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+
+    Herbivore.alive.push(this);
```

Comment:
[CODE PERFOMANCE] - Instead of pushing the new instance to `Herbivore.alive`, you should add it to `Animal.alive`. This way, you maintain a single source of truth for all alive animals, and you avoid issues with inheritance since `Herbivore` does not have its own `alive` static property.